src/main/java/com/google/sps/servlets/MapServlet.java

```diff
@@ -1,54 +1,48 @@
 package com.google.sps.servlets;
 
-import com.google.template.soy.SoyFileSet;
-import com.google.template.soy.tofu.SoyTofu;
+import static com.google.sps.utils.ServletUtils.getGroupsList;
+import static com.google.sps.utils.SoyRendererUtils.getOutputString;
+import static com.google.sps.utils.StringConstants.*;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
+import com.google.sps.utils.ServletUtils;
 import java.io.File;
 import java.io.IOException;
+import java.util.HashSet;
 import java.util.Scanner;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-
-
+ 
+ 
 @WebServlet("/map")
 public class MapServlet extends HttpServlet {
-
+ 
     /**
      * Adds user's groups as checkboxes to the map drawer
      * Allows users to select which groups they want displayed on the map
      */
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-
+        ServletUtils.enforceUserLogin(request, response);
+ 
         // Reads API Key and HTTP referer from file
```

Comment:
you might want to change your comment here, since you're no longer reading the referer